source/library/Witch/Instances.hs

```diff
@@ -1334,6 +1336,56 @@ instance From.From String (Encoding.UTF_8 ByteString.ByteString) where
 instance From.From String (Encoding.UTF_8 LazyByteString.ByteString) where
   from = Utility.via @LazyText.Text
 
+-- UTF-16LE
+
+-- | Uses 'Text.decodeUtf16LE'.
+instance TryFrom.TryFrom (Encoding.UTF_16LE ByteString.ByteString) Text.Text where
+  tryFrom = Utility.eitherTryFrom $ tryEvaluate @Text.UnicodeException . Text.decodeUtf16LE . From.from
+
+-- | Converts via 'Text.Text'.
+instance TryFrom.TryFrom (Encoding.UTF_16LE ByteString.ByteString) LazyText.Text where
+  tryFrom = Utility.eitherTryFrom $ fmap (Utility.into @LazyText.Text) . Utility.tryInto @Text.Text
+
+-- | Converts via 'Text.Text'.
+instance TryFrom.TryFrom (Encoding.UTF_16LE ByteString.ByteString) String where
+  tryFrom = Utility.eitherTryFrom $ fmap (Utility.into @String) . Utility.tryInto @Text.Text
+
+-- | Uses 'LazyText.decodeUtf16LE'.
+instance TryFrom.TryFrom (Encoding.UTF_16LE LazyByteString.ByteString) LazyText.Text where
+  tryFrom = Utility.eitherTryFrom $ tryEvaluate @Text.UnicodeException . LazyText.decodeUtf16LE . From.from
```

Comment:
I suspect that this is not lazy enough. I haven't checked, but I imagine if you try to take a valid prefix from a UTF-8 encoded lazy byte string, it would work whereas this UTF-16LE variant would fail.

Reply:
Oh wait, never mind:

``` hs
ghci> fmap (LT.take 1) . LT.decodeUtf8' $ LB.pack [0x21, 0xff]
Left Cannot decode byte '\xff': Data.Text.Internal.Encoding.streamDecodeUtf8With: Invalid UTF-8 stream
```

And of course that makes sense. You can't know if the result is `Left` or `Right` without parsing the entire input. Maybe these encoding instances should push everything through strict `Text` for efficiency? Although I'm not actually sure if that would be more efficient.